lib/ruby_indexer/lib/ruby_indexer/declaration_listener.rb

```diff
@@ -9,7 +9,7 @@ class DeclarationListener
     #: Array[String]
     attr_reader :indexing_errors
 
-    #: (Index index, Prism::Dispatcher dispatcher, Prism::ParseResult parse_result, URI::Generic uri, ?collect_comments: bool) -> void
+    #: (Index index, Prism::Dispatcher dispatcher, Prism::ParseLexResult | Prism::ParseResult parse_result, URI::Generic uri, ?collect_comments: bool) -> void
```

Comment:
Do we really want to continue accepting bare `ParseResult`?

```suggestion
    #: (Index index, Prism::Dispatcher dispatcher, Prism::ParseLexResult parse_result, URI::Generic uri, ?collect_comments: bool) -> void
```

Reply:
Yes, because `parse_lex` is slower than `parse` and we only need it for RuboCop AST translations. I don't want to slow down initial indexing since that has no need for the lex.